Add support for buildkite_agent_count (#43)
improbable-eng/ansible-buildkite-agent: CHANGELOG.md

## Unreleased

> ## 3.2.1: 2020-11-18
> 

Reviewer: ```suggestion
```

I'll fill in the release version and date after merge and an internal test run if that's ok.

> Add support to configure number of agents to spawn with launchd on macOS by setting the --spwan flag.

Reviewer: ```suggestion
Fix: macOS can run `buildkite_agent_count` job-runners now.
```

Author: Not sure if this is correct with my current solution though. `buildkite_agent_count` is still not used but instead the new variable introduced in this PR for `buildkite_agent_spawn`. Like #42 states `buildkite_agent_count` isn't used (on any OS) so maybe just remove that parameter?

Reviewer: Oh, I see! Sorry, I misunderstood. In that case, yes, if you wouldn't mind swapping over to `buildkite_agent_spawn` throughout (as in, this PR, I'll do the other two platforms at some stage) and dropping `_count`, that would be very welcome.

Reviewer: My code suggestions are based on thinking `buildkite_agent_count` existed and was active; please replace to `buildkite_agent_spawn` instead.

Author: Nono, you're right, I'm the one not fully understanding here. To be honest I'm pretty new with both Ansible and Buildkite so it's a bit of cargo cult programming. I appreciate your help and feedback suggestions! The config `buildkite_agent_count` exists but isn't used. Setting it's value to the variable `BUILDKITE_AGENT_SPAWN` which will then be used by Buildkite to configure `--spawn` seems like a good idea.

Or actually, reading the docs, this could be set just by adding `spawn` to the Buildkite configuration. And by doing so we woudn't even have to manage the plist file at all. Let me try that!

Author: Yep, my bad. This is basically all that the PR needed for this to work:

```diff
diff --git a/templates/buildkite-agent.cfg.j2 b/templates/buildkite-agent.cfg.j2
index e18622b..239e986 100644
--- a/templates/buildkite-agent.cfg.j2
+++ b/templates/buildkite-agent.cfg.j2
@@ -4,6 +4,9 @@ token="{{ buildkite_agent_token }}"
 # The name of the agent
 name="%hostname-%n"

+# Number of agents to spawn
+spawn="{{ buildkite_agent_count | default('1') }}"
+
```

Now I feel stupid, but I guess this is better. Will push these changes, if you still want to manage a plist file let me know!

> * Added: `buildkite_agent_spawn` (which will default to env `BUILDKITE_AGENT_SPAWN`)

Reviewer: ```suggestion
```

## Unreleased

> Fix: macOS can run `buildkite_agent_count` job-runners now.

Reviewer: ```suggestion
* Fix: all platforms will run `buildkite_agent_count` job-runners now, defaulting to `1`.
* Fix: macOS `buildkite_agent_executable` is correct for the homebrew install now.
```

Author: Heh, I just pushed before reading this! Let's use this one!
